Source/SwiftLintFramework/Rules/VerticalWhitespaceBetweenCasesRule.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+public struct VerticalWhitespaceBetweenCasesRule {
+    public init() {}
+
+    private static let violatingToValidExamples: [String: String] = [
+    """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+        default:
+            print("x is invalid")
+        }
+    """: """
+        switch x {
+        case 0..<5:
+            print("x is valid")
+
+        default:
+            print("x is invalid")
+        }
+    """
+    ]
+
+    private let pattern = "([^\\n{][ \\t]*\\n)([ \\t]*(?:case[^\\n]+|default):[ \\t]*\\n)"
+}
+
+extension VerticalWhitespaceBetweenCasesRule: OptInRule {
+    public var configurationDescription: String { return "N/A" }
+
+    public init(configuration: Any) throws {}
+
+    public static let description = RuleDescription(
+        identifier: "vertical_whitespace_between_cases",
+        name: "Vertical Whitespace Between Cases",
+        description: "Include a vertical whitespace (empty line) between cases in switch statements.",
+        kind: .style,
+        nonTriggeringExamples: Array(Set(violatingToValidExamples.values)),
+        triggeringExamples: Array(Set(violatingToValidExamples.keys)),
+        corrections: violatingToValidExamples
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        return file.violatingRanges(for: pattern).map {
+            StyleViolation(
+                ruleDescription: type(of: self).description,
+                severity: ViolationSeverity.warning,
```

Comment:
Most rules allow developers to set the severity of the violations triggered. One might want to set it to error instead or warning.

Reply:
Done.